smartsim/_core/launcher/step/localStep.py

```diff
@@ -39,7 +40,9 @@ def __init__(self, name: str, cwd: str, run_settings: RunSettings):
 
     @property
     def run_settings(self) -> RunSettings:
-        return self.step_settings
+        if isinstance(self.step_settings, RunSettings):
+            return self.step_settings
+        raise TypeError("Run settings must be of type RunSettings")
```

Comment:
Codecov is complaining that the added line is not covered by tests. Do you think it is worth to add tests for this and all other new lines which are not covered? Or should we disable coverage for them?